### ISSUE TITLE
Different Categories for CLS dataset for French 

### ISSUE THREAD

**Issue by yongzx** [getalp/Flaubert#36]
I am using Huggingface datasets for FLUE benchmark, and the CLS datasets are grouped together instead of being separated into different categories of "Books", "DVD", and "Music". 

What's the best way of splitting the CLS datasets into different subsets?

**Comment by formiel** [comment]
Hi @yongzx 

Thank you so much for your interest in FLUE! I'm not familiar with HuggingFace's `datasets` but I think we can get separate categories by doing something like this.
- Add a field for `category` in [line 107](https://github.com/huggingface/datasets/blob/03b7f123cc17afc517c0aa2f912bbd90cb266185/datasets/flue/flue.py#L107): `text_features={"text": "text", "category": "category"},`
- And add information of the `category` when generating the examples in [line 390](https://github.com/huggingface/datasets/blob/03b7f123cc17afc517c0aa2f912bbd90cb266185/datasets/flue/flue.py#L390): `yield f"{category}_{id_}", {"idx": id, "text": review_text, "label": label, "category": category}`. 

Unfortunately for some reason it’s not fully working and I’m not sure why. But hopefully this can help in some way. Please let me know if you can make it work.

**Comment by yongzx** [comment]
Actually this works!!

The code would be
```
cls_train_datasetdict = load_dataset("/path/to/dir/flue", "CLS", 
                                     split=f"train", 
                                     cache_dir="/path/to/dir/data/external/flue-cls").train_test_split(train_size=0.8, shuffle=True, seed=42)
```
where `/path/to/dir/flue` contains the modified `flue.py`. Thanks!!
